### PR TITLE
SLING-8591 - Hook to support custom code when installing or deleting …

### PR DESCRIPTION
**Pull request by cschneider**
…distribution packages

### CODE REVIEW

File: src/test/java/org/apache/sling/distribution/packaging/impl/InMemoryDistributionPackageBuilderTest.java [hunk 1]
```diff
@@ -18,30 +18,97 @@
...
+        InputStream inputStream = new ByteArrayInputStream(outputStream.toByteArray());
+
+        DistributionPackage distPackage = builder.readPackage(resolver, inputStream);
+        DistributionPackageInfo outInfo = distPackage.getInfo();
+        assertThat(outInfo.get("test1", String.class), equalTo("value1"));
+    }
+    
+    @Test
+    public void testGetPackageInvalid() throws Exception {
+
```
Comment (tmaret) [inline src/test/java/org/apache/sling/distribution/packaging/impl/InMemoryDistributionPackageBuilderTest.java]: sonar hint, seems empty :-)

File: src/test/java/org/apache/sling/distribution/queue/impl/resource/ResourceQueueIT.java [hunk 2]
```diff
@@ -48,6 +49,7 @@
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
+@Ignore
```
Comment (tmaret) [inline src/test/java/org/apache/sling/distribution/queue/impl/resource/ResourceQueueIT.java]: How does ignoring this test relate to SLING-8591 ?

File: src/test/java/org/apache/sling/distribution/packaging/impl/InMemoryDistributionPackageBuilderTest.java (hunk 1 above)
Comment (cschneider) [reply src/test/java/org/apache/sling/distribution/packaging/impl/InMemoryDistributionPackageBuilderTest.java]: Yes .. actually a good catch by sonar .. I moved the test to a different class and forgot to remove this one.

File: src/test/java/org/apache/sling/distribution/queue/impl/resource/ResourceQueueIT.java (hunk 2 above)
Comment (cschneider) [reply src/test/java/org/apache/sling/distribution/queue/impl/resource/ResourceQueueIT.java]: Also a leftover .. When I run the tests in eclipse this test kind of blocks .. so I have to ignore it ... and not forget to remove the ignore on commit. Actually if you have the chance maybe you could check if the test also blocks for you inside eclipse or do you use intellij?

**Pull request merged**